torch/optim/nadam.py

```diff
@@ -299,24 +308,38 @@ def _multi_tensor_nadam(params: List[Tensor],
                         momentum_decay: float,
                         eps: float,
                         decoupled_weight_decay: bool,
+                        capturable: bool,
                         differentiable: bool):
 
     if len(params) == 0:
         return
 
     assert not differentiable, "_foreach ops don't support autograd"
 
-    grouped_tensors = Optimizer._group_tensors_by_device_and_dtype([params, grads, exp_avgs, exp_avg_sqs, mu_products, state_steps])
+    grouped_tensors = Optimizer._group_tensors_by_device_and_dtype(
+        [params, grads, exp_avgs, exp_avg_sqs, mu_products, state_steps])
     for ((grouped_params, grouped_grads, grouped_exp_avgs,
          grouped_exp_avg_sqs, grouped_mu_products, grouped_state_steps), _) in grouped_tensors.values():
 
         # update steps
         torch._foreach_add_(grouped_state_steps, 1)
 
-        bias_correction2 = [1 - beta2 ** _get_value(step) for step in grouped_state_steps]
-        mus = [beta1 * (1. - 0.5 * (0.96 ** (_get_value(step) * momentum_decay))) for step in grouped_state_steps]
-        mu_nexts = [beta1 * (1. - 0.5 * (0.96 ** ((_get_value(step) + 1) * momentum_decay)))
-                    for step in grouped_state_steps]
+        # scalar python math is faster in eager
+        # however, it is not visible to cudagraphs
+        # and can be slow to trace
```

Comment:
Oh, I see....

If you're introducing a new branch of implementation, then it will have to be tested for:
- correctness (does it output the right results?)
- minimizing use of peak memory (see test_peak_memory for foreach).

Could you give some stats on speed differences here? It would be good to quantify the exact tradeoff of adding another branch.